bot/converters.py

```diff
@@ -574,6 +574,55 @@ async def convert(self, ctx: Context, arg: str) -> t.Optional[dict]:
                 raise e
 
 
+class DayDuration(Converter):
+    """
+    Convert a string representing day time (hours and minutes) to a UTC datetime object.
+
+    The hours and minutes would be combined with UTC day. If no 'am' or 'pm' is passed with
+    the string, then it is assumed that the time is in 24 hour format.
+
+    The following formats are accepted:
+        - H:M
+        - H:Mam/pm
+        - HMam/pm
+        - Ham/pm
+        - H
+
+    where `H` represents Hours and `M` represents Minutes.
+    """
+
+    TIME_RE = re.compile(
+        r"^(1[0-2]|0?[1-9]):?([0-5][0-9])? ?([AaPp][Mm])$"  # Twelve hour format
+        "|"
+        r"^([0-9]|0[0-9]|1[0-9]|2[0-4]):?([0-5][0-9])?$"   # Twenty four hour format
```

Comment:
Taking a look at the regex, `24:xx` isn't a valid 24hr time so the `|2[0-4]` should actually be `|2[0-3]`.

Reply:
```py
TIME_RE = re.compile(
    r"(?P<hour_12>0?[1-9]|1[0-2])"  # hours (12 hour format)
    r"|"
    r"(?P<hour_24>00?|1[2-9]|2[0-3]))"  # hours (24hr format)
    r"[:.]"  # separator between hours and minutes is `:` or `.`
    r"(?P<minute>[0-5][0-9])"  # minutes
    r"(?::"  # separator between minutes and seconds is `:`
    r"(?P<second>[0-5][0-9]))?"  # optional seconds
    r"(?: ?)"  # separator between minutes/seconds and meridiem is a space or nothing
    r"(?P<meridiem>[AaPp][Mm])?"  # meridiem
)
```
perhaps something like this, formatted however you want (I did extra new lines here to make it more readable) - although note that this requires validation that meridiem wasn't passed for a 24 hour time